File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        let start = 0;
        let end = numbers.length - 1;
        let res = [];

        while (start < end) {
            let temp = numbers[start] + numbers[end];
            if (temp == target){
                res.push(start + 1);
                res.push(end + 1);
                break;
            } else if (temp < target) {
                start++;
            } else {
                end--;
            }
        }

        return res;
    }
}

File: Data Structures & Algorithms/car-fleet/submission-1.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        vector<vector<int>> tups{};
        vector<double> res{};

        for (int i=0; i<position.size(); i++){
            tups.push_back({position[i], speed[i]});
        }

        sort(tups.begin(), tups.end());
        
        for (int i=tups.size() - 1; i>=0; i--) {
            double time = (double)(target - tups[i][0]) / tups[i][1];
            if (res.size() == 0){
                cout << time << "\n";
                res.push_back(time);
            } else {
                double temp = res.back();
                cout << time << ":" << temp << "\n";
                if (temp < time){
                    res.push_back(time);
                }
            }
        }


        return res.size();
    }
};
